clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3221,6 +3223,17 @@ protected void processEndOfPush(
      * Generate snapshot after batch write is done.
      */
     if (storeVersionConfig.isBlobTransferEnabled() && serverConfig.isBlobTransferManagerEnabled()) {
+      // 1. Add the pre-snapshot listener to partition for hybrid stores. No need to listen to batch stores.
+      if (isHybridMode()) {
+        try {
+          addBlobTransferSnapshotCreationListener(storageEngine, partition);
+        } catch (Exception e) {
+          LOGGER
+              .warn("Failed to setup pre-snapshot listener for topic {} partition {}", kafkaVersionTopic, partition, e);
+        }
+      }
+
+      // 2. Create snapshot for blob transfer
       storageEngine.createSnapshot(storagePartitionConfig);
```

Comment:
So we will create a snapshot for hybrid store in Server all the time when this feature is enabled?
Who will clean this up if there is no blob transfer request?
Unused checkpoint can keep the stale data lingering around.